overlay/files/system/oem/05_network.yaml

```diff
@@ -13,6 +13,14 @@ stages:
       commands:
         - rm /etc/resolv.conf
         - ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf
+# This change ensures any network config files updated during initramfs are read into systemd-networkd and the stack is restarted. This fixes issues with the DHCP IP address getting dropped under certain circumstances in Equinix Metal. Also fixes dynamically defined network bonding not coming up properly until the daemon-reload and restart commands are issued. See also: https://github.com/kairos-io/kairos/pull/1030
+  boot.after:
+    - name: "Restart network"
+      commands:
+        - systemctl daemon-reload
+        - systemctl restart systemd-networkd
+        - |
```

Comment:
I seem to fail to suggest the change, but this should be : 

```yaml
 boot.after:
    - name: "Restart network"
      if: ...
```

Reply:
It seems the `boot` stage doesn't get executed during a live boot, at least I didn't get the proper result when using `boot.after` for some servers. Switching back to `initramfs.after` turns out to be more reliable for the live mode boot. I'll gate the entire step then to only be executed during live mode boot as you suggested, and use a seperate network-stack restart in the boot stage as part of the cloud-init configuration.